Finjector.Web/ClientApp/src/components/Details/DetailsChartString.tsx

```diff
@@ -5,10 +5,10 @@ import { Badge } from "reactstrap";
 import usePopupStatus from "../../util/customHooks";
 
 interface DetailsChartStringProps {
-  chartType: string;
-  chartString: string;
-  isValid: boolean;
-  hasWarnings: boolean;
+  chartType: string | undefined;
```

Comment:
is doing `chartType: string | undefined;` the same as doing `chartType?: string;`?  If so i prefer the later.

Reply:
no, `chartType: string | undefined` makes it required as a prop but allows that prop to be undefined (like before the data has loaded). `chartType?: string` makes it an optional prop, and chartType is necessary for rendering this component. not really an issue in this case but i'd rather make it future-fool proof